stdlib/LinearAlgebra/src/LinearAlgebra.jl

```diff
@@ -468,6 +468,10 @@ _evview(S::SymTridiagonal) = @view S.ev[begin:length(S.dv) - 1]
 _zeros(::Type{T}, b::AbstractVector, n::Integer) where {T} = zeros(T, max(length(b), n))
 _zeros(::Type{T}, B::AbstractMatrix, n::Integer) where {T} = zeros(T, max(size(B, 1), n), size(B, 2))
 
+# append a zero element / drop the last element
+_pushzero(A) = (B = similar(A, length(A)+1); B[begin:end-1] .= A; B[end] = zero(eltype(B)); B)
```

Comment:
```suggestion
```
You could equally well just write `[A, zero(eltype(A))]`. This will also allocate new memory and not modify the diagonals.

Reply:
True.